artifacts/Invoke-CommandInPwshCore.ps1

```diff
@@ -0,0 +1,81 @@
+function Invoke-CommandInPwshCore() {
+    # Must be a simple function for correct splatting
+    Param(
+        [scriptblock]$ScriptBlock,
+    
+        [Alias('db')][switch]$Debug,
+        [Alias('vb')][switch]$Verbose,
+        [Alias('ea')][string]$ErrorAction,
+        [Alias('ev')][string]$ErrorVariable,
+        [Alias('infa')][string]$InformationAction,
+        [Alias('iv')][string]$InformationVariable,
+        [Alias('ob')][int]$OutBuffer,
+        [Alias('ov')][string]$OutVariable,
+        [Alias('pv')][string]$PipelineVariable,
+        [Alias('pa')][string]$ProgressAction,
+        [Alias('wa')][string]$WarningAction,
+        [Alias('wv')][string]$WarningVariable
+    )
+    $commonCmdLetParams = [hashtable]$PSBoundParameters
+    $commonCmdLetParams.Remove('ScriptBlock');
+
+    if ($PSVersionTable.PSEdition -eq 'Core') {
+        & $ScriptBlock @args @commonCmdLetParams
+        return
+    }
+
+    $pwshCoreSessionConfigurationName = "PowerShell.7"
+    $pwshCoreSessionName = "PwshCoreSession"
+
+    if (! $pwshCoreSession) {
+        # Find existing session (open or disconnected)
+        $pwshCoreSession = Get-PSSession -Name $pwshCoreSessionName -ea silentlycontinue | Where-Object { $_.State -in @("Opened","Disconnected") } | Select-Object -Last 1
+    }
+
+    if (! $pwshCoreSession) {
+        # Check powershell core exists
+        if (! (Get-Command pwsh -ea SilentlyContinue)) {
+            throw "Powershell core not found"
```

Comment:
maybe use "normal" PS then? I'm not 100% sure if all BC versions really have PS core available

Reply:
But would it not be better to have an error instead of the hang-up? 

As of BC24 the containers should contain PS core because they contain the new module "Microsoft.BusinessCentral.Apps.Management" that only works in core and the override of Publish-NAVApp only forwards to PS core if this module exists in the filesystem.

Reply:
the hung-up only happens when the publish fails (e.g. due to compile errors). When trying with the non-PS core we have a chance that it works anyway (when no errors happen).

Your changes apply to every BC version and not just BC24+, right? If they apply to all of them I would suggest a "Fallback" to normal PS is PS Core is not available but at least print a warning that it might hang

Reply:
Sure but the worst case would be a hang-up because I get compile errors (e.g. in next Major/Minor pipelines) and in this case the user would be happier if he got a real error instead of a timeout (imho).

The override itself applies for every BC version but will only forward to PS Core for BC24+.

If you think a fallback would be better, than I can implement it but the warning at this position would be more generic because this is the forward-function that can be used for more than Publish-NAVApp

Reply:
As an alternative I could also install pwsh if it missing.

Reply:
Discussed in teams: No fallback needed, because PS Code will only be used for BC24+ and onwards.